todoapp/src/components/Todo.tsx

```diff
@@ -28,54 +29,77 @@ const Todos = () => {
   const { tasksList } = useGlobalState();
   const text = useHookstate("");
   const selectedTodoState = useHookstate<string | null>(null);
+  const crudAPI = "https://crudcrud.com/api/8294360fe502453db515d4bc78dae699/todos"
+
+  useEffect(() => {
+    fetchTodos();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const fetchTodos = async () => {
+    try {
+      const response = await axios.get(crudAPI);
+      const todos = response.data;
+      tasksList.set(todos);
+    } catch (error) {
+      console.error(error);
+    }
+  };
 
   const handleChange = (value: string) => {
     text.set(value);
   }
-  const createTodo = () => {
+  const createTodo = async () => {
     const newText = text.get().trim()
-    const prevTodos = tasksList.get();
-    if (selectedTodoState.value && prevTodos) {
-      const updatedTodos = prevTodos.map((todo) => {
-        if (todo.id === selectedTodoState.value) {
-          return { ...todo, text: newText };
-        }
-        return todo;
-      });
-
-      tasksList.set(updatedTodos);
-      selectedTodoState.set(null);
-    } else {
-      const newTodo: Todo = {
-        id: Date.now().toString(),
-        text: newText,
-        completed: false,
-      };
-      tasksList.set((prevTodos: Todo[]) => [...prevTodos, newTodo]);
+    const newTodo = {
+      text: newText,
+      completed: false,
+    };
+    try {
+      const response = await axios.post(crudAPI, newTodo);
+      const createdTodo = response.data;
+      tasksList.set((prevTodos: Todo[]) => [...prevTodos, createdTodo]);
+      text.set("");
+      fetchTodos()
+    } catch (error) {
+      console.error(error);
     }
-
-    text.set('');
   };
 
-  const deleteTodo = (id: string) => {
-    tasksList.set((prevTodos: Todo[]) =>
-      prevTodos.filter((todo) => todo.id !== id)
-    );
+  const deleteTodo = async (id: string) => {
+    try {
+      await axios.delete(`${crudAPI}/${id}`);
+      tasksList.set((prevTodos: Todo[]) =>
+        prevTodos.filter((todo) => todo._id !== id)
+      );
```

Comment:
instead of this call fetchTodos() function and it will set the updated tasklist itself by calling an api